### PR TITLE
Refactor HiFi-GAN Conv1d padding with `same` arg

### LINKED ISSUE #395: Refactor Conv1d's `padding` with `same`
## Summary
Current HiFi-GAN implementation calculate `padding` manually in `Conv1d`.  
We can refactor them with `same` argument.  
This can make the codes intuitive, and enable future big refactoring of `use_causal_conv`-related redundant codes.  
I made a pull request.  

## Current Status
Current HiFi-GAN implementation calculate `padding` of  `Conv1d(stride=1)` as `(kernel_size - 1) // 2`.  
This manual calculation intend to keep signal length (time dimension size).  
https://github.com/kan-bayashi/ParallelWaveGAN/blob/ffaa99fe77d3b0703e5857177fd9b2ecc18cb0bd/parallel_wavegan/models/hifigan.py#L80

But, PyTorch already provide argument `padding="same"` for same purpose.  
>  padding='same' pads the input so the output has the shape as the input.  ([ref](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html#conv1d)).  

So, current implementation has redundant manual calculation.  
There are 9 redundant parts.  

## Proposal
Just replace `padding = (kernel_size - 1) // 2` with `"same"`.  
HiFi-GAN already guarantee 'k is odd' with assert, so there is no behavior change.  
So, this is not breaking change, but just refactoring.  
 
## Merits
We can get two advange with this refactoring.  

- clean code
- future big refactoring

At first, we can replace manual calculation with intuitive "same" string. Code becomes clean.  

More importantly, this opens the opportunity to remove redundant if/else caused by `use_causal_conv`.  
HiFi-GAN has many if/else, and it is caused by `padding` argument in Causal mode.  
Some refactoring after `padding="same"` can delete these if/else (totally 50~ line reduction).  

## PR
I made the pull request (#396).

### PR DESCRIPTION
**Pull request by tarepan**
resolve #395

### CODE REVIEW

**Comment by tarepan** [comment]
Test failed because of PyTorch version. Old PyTorch do not support `same` argument.  

I have forgotten this fact...  
Are there any policy about PyTorch version in this repository?  
For example, "support PT1.4 as long as we can".

**Comment by kan-bayashi** [comment]
@tarepan Thank you for your kind summary and PR. padding=same is useful, but it is available in only recent pytorch. I want to keep the compatibility with older version. Therefore, I could not merge this PR. Anyway, Thank you for your good suggestions.

**Comment by kan-bayashi** [comment]
It seems padding same is available from 1.8.

**Comment by kan-bayashi** [comment]
> Are there any policy about PyTorch version in this repository?

At least, I want to keep the old version which supported in espnet.

**Comment by tarepan** [comment]
I understand the situation.  
Thanks for your kind review!
